edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -256,3 +256,101 @@ def test_exams_config(self, other_course_id, exam_type, expect_none_provider):
+
+        self.course_id = 'course-v1:edx+test+f19'
+
+        self.url = reverse('api:v1:exams-course_exam_configs', kwargs={'course_id': self.course_id})
+
+    # todo: can this helper method be moved to base?
+    def patch_api(self, user, data):
+        """
+        Helper function to make patch request to the API
+        """
+
+        data = json.dumps(data)
+        headers = self.build_jwt_headers(user)
+
+        return self.client.patch(self.url, data, **headers, content_type="application/json")
+
+    # todo: can this helper method be moved to base since both views use it?
+    def get_response(self, user, data, expected_response):
+        """
+        Helper functions to get API response
+        """
+        response = self.patch_api(user, data)
+        self.assertEqual(response.status_code, expected_response)
+
+        return response
+
+    def test_auth_failures(self):
+        """
+        Verify the endpoint validates permissions
+        """
+        # Test unauthenticated
+        response = self.client.patch(self.url)
+        self.assertEqual(response.status_code, 401)
+
+        # Test non-staff user
+        random_user = UserFactory()
+        self.get_response(random_user, {}, 403)
+
+    def test_invalid_data(self):
+        """
+        Assert that endpoint returns 400 if provider is missing
+        """
+        data = {}
+        self.get_response(self.user, data, 400)
+
+    def test_invalid_provider(self):
+        """
+        Assert endpoint returns 400 if provider is invalid
+        """
+        data = {
+            'name': 'test_provider',
+            'lti_configuration_id': '123456789',
+        }
+        response = self.get_response(self.user, data, 400)
+        self.assertIn("verbose_name", response.data["errors"])
+
+    def test_config_update(self):
+        """
+        Test that config is updated
+        """
+        config = CourseExamConfiguration.objects.create(course_id=self.course_id,
+                                                        provider=self.test_provider)
+        config.save()
+
+        data = {
+            'name': 'test_provider2',
+            'verbose_name': 'testing_provider_2',
+            'lti_configuration_id': '223456789'
+        }
+
+        self.assertEqual(len(CourseExamConfiguration.objects.all()), 1)
+        self.get_response(self.user, data, 200)
+        self.assertEqual(len(CourseExamConfiguration.objects.all()), 1)
+        # todo: should I test that the config is correct? anything else to test?
+
+    def test_config_create(self):
+        """
+        Test that config is created
+        """
+        data = {
+            'name': 'test_provider',
+            'verbose_name': 'testing_provider',
+            'lti_configuration_id': '123456789'
+        }
+        self.assertEqual(len(CourseExamConfiguration.objects.all()), 0)
+        self.get_response(self.user, data, 200)
+        self.assertEqual(len(CourseExamConfiguration.objects.all()), 1)
```

Comment:
I think you should test that your desired data is actually what was saved. Otherwise, this test can only tell you that _something_ was created. The only way to ensure that the `CourseExamConfiguration` that you expected was created was to test the values. For example, you could test that the `provider_id` field maps to the correct `ProctoringProvider`.

Reply:
@MichaelRoytman cool! I ended up checking that the providers are the same as expected.